go/types/indexed_sequences.go

```diff
@@ -77,3 +80,74 @@ func orderedKeyFromSum(msd []metaTuple) orderedKey {
 	}
 	return orderedKeyFromUint64(sum)
 }
+
+func LoadLeafSequences(c Collection, startIdx, endIdx uint64) (leaves []sequence, firstIdx uint64) {
+	return loadLeafSequences(c.sequence().valueReader(), []sequence{c.sequence()}, startIdx, endIdx)
+}
+
+func loadLeafSequences(vr ValueReader, seqs []sequence, startIdx, endIdx uint64) ([]sequence, uint64) {
+	if seqs[0].isLeaf() {
+		for _, s := range seqs {
+			d.PanicIfFalse(s.isLeaf())
+		}
+
+		return seqs, startIdx
+	}
+
+	level := seqs[0].treeLevel()
+	childTuples := []metaTuple{}
+
+	cum := uint64(0)
+	for _, s := range seqs {
+		d.PanicIfFalse(s.treeLevel() == level)
+		ms := s.(metaSequence)
+
+		for _, mt := range ms.tuples {
+			if cum == 0 && mt.numLeaves <= startIdx {
```

Comment:
it would have helped to have had a comment above here like "skip tuples up to starting index"